pqarrow/builder/recordbuilder.go

```diff
@@ -117,3 +117,14 @@ func (b *RecordBuilder) ExpandSchema(schema *arrow.Schema) {
 
 	b.schema = schema
 }
+
+// Reset will call ResetFull on any dictionary builders to prevent memo tables from growing unbounded.
+func (b *RecordBuilder) Reset() {
+	for _, f := range b.fields {
+		if lb, ok := f.(*ListBuilder); ok {
```

Comment:
Do we also want to do this on dictionary builders that are not wrapped by a list builder?

Reply:
So, we only see this issue with the repeated dictionaries, whereas the other dictionaries probably benefit from not a full reset, but I don't know how to express that.